inc/user.class.php

```diff
@@ -4271,15 +4339,19 @@ static function showPasswordForgetChangeForm($token) {
 
       // Verif token.
       $token_ok = false;
-      $query = "SELECT *
-                FROM `glpi_users`
-                WHERE `password_forget_token` = '$token'
-                      AND NOW() < ADDDATE(`password_forget_token_date`, INTERVAL 1 DAY)";
-
-      if ($result = $DB->query($query)) {
-         if ($DB->numrows($result) == 1) {
-            $token_ok = true;
-         }
+      $iterator = $DB->request([
+         'FROM'   => self::getTable(),
+         'WHERE'  => [
+            'password_forget_token'       => $token,
+            new \QueryExpression('NOW()') => [
```

Comment:
Not sur a `QueryExpression` could be used as array key
```diff
-            new \QueryExpression('') => [
-               '<',
-               new \QueryExpression('ADDDATE(password_forget_token_date, INTERVAL 1 DAY))')
-            ]
+           new \QueryExpression('NOW() < ADDDATE(password_forget_token_date, INTERVAL 1 DAY))')
```

Reply:
It should be; but anyways the way you propose is more readable.